libcore/ConnectServerDialog.vala

```diff
@@ -0,0 +1,530 @@
+/*
+* Copyright (c) 2015-2018 elementary LLC. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1335 USA.
+*
+* Authored by: Corentin Noël <corentin@elementary.io>
+*/
+
+public class PF.ConnectServerDialog : Gtk.Dialog {
```

Comment:
Could avoid some layout work with [`Granite.MessageDialog`](https://valadoc.org/granite/Granite.MessageDialog.html)

Reply:
I'll try it, thanks.